src/main/scala/is/hail/methods/LinearMixedRegression.scala

```diff
@@ -55,50 +61,105 @@ object LinearMixedRegression {
 
     val covNames = "intercept" +: covExpr
 
-    val filteredKinshipMatrix = if (kinshipMatrix.sampleIds sameElements completeSamples)
-      kinshipMatrix
-    else {
-      val fkm = kinshipMatrix.filterSamples(completeSamplesSet)
-      if (!(fkm.sampleIds sameElements completeSamples))
-        fatal("Array of sample IDs in assoc_vds and array of sample IDs in kinship_matrix (with both filtered to complete " +
-          "samples in assoc_vds) do not agree. This should not happen when kinship_matrix is computed from a filtered version of assoc_vds.")
-      fkm
-    }
-
     val n = y.length
     val k = cov.cols
     val d = n - k - 1
 
     if (d < 1)
-      fatal(s"$n samples and $k ${ plural(k, "covariate") } including intercept implies $d degrees of freedom.")
+      fatal(s"lmmreg: $n samples and $k ${plural(k, "covariate")} including intercept implies $d degrees of freedom.")
+
+    info(s"lmmreg: running lmmreg on $n samples with $k sample ${plural(k, "covariate")} including intercept...")
+
+    //TODO This whole match thing got a little ugly as complexity increased. Consider cleaning up.
+    val (u, s, rank, fullS): (DenseMatrix[Double], DenseVector[Double], Int, DenseVector[Double]) = similarityMatrix match {
+      case LDMatrix(irm, variants, nSamplesUsed) => {
+        val variantSet = variants.toSet
+        val ldDim = irm.numCols().toInt
+        val localMatrix = irm.toLocalMatrix().asBreeze().toDenseMatrix
+
+        info(s"lmmreg: Computing eigenvectors of LD matrix...")
+        val eigK = printTime(eigSymD(localMatrix))
+        info(s"lmmreg: Eigendecomposition complete")
+        val fullV = eigK.eigenvectors
+
+        val c1 = nSamplesUsed.toDouble / variants.length
+        val fullS = eigK.eigenvalues.map(e => e * c1)
+        val fullNEigs = fullS.length
+
+        val rank = min(variants.length, nSamplesUsed)
+
+        val nEigs = computeNEigs(fullS, optNEigs, optDroppedVarianceFraction, rank)
+
+        val S = fullS((fullNEigs - nEigs) until fullNEigs)
+        val V = fullV(::, (fullNEigs - nEigs) until fullNEigs)
+
+        val c2 = 1.0 / math.sqrt(variants.length)
+        val sqrtSInv = S.map(e => c2 / math.sqrt(e))
+
+        var filteredVDS = vds.filterVariants((v, _, _) => variantSet(v))
```

Comment:
So the problem here is similar to the problem I addressed in your first comment. You want to be able to run the per variant test on variants that are not in the LD matrix. So the VDS should be able to have non-LD-matrix variants in it, but you only what the ones in the VDS that are also in the LD Matrix to make the genotype matrix to multiply by